src/Autosuggest.js

```diff
@@ -89,6 +95,17 @@ class Autosuggest extends Component {
     }
   }
 
+  componentWillUnmount() {
+    global.window.removeEventListener('mouseup', this.onDocumentMouseUp, false);
+    global.window.removeEventListener('touchend', this.onDocumentMouseUp, false);
+  }
+
+  onDocumentMouseUp() {
+    setTimeout(() => {
+      this.justClickedOnSuggestion = false;
+    });
```

Comment:
yes, in order for `justClickedOnSuggestion` to be set when `inputProps.onBlur` is called, we still need a timeout.